src/playlist-maker.js

```diff
@@ -0,0 +1,57 @@
+var playItem = require('./playitem.js');
+var setupAutoadvance = require('./autoadvance.js');
+
+var playlistMaker = function(player, plist) {
+  var currentIndex = 0;
+  var autoadvanceTimeout = null;
+
+  var playlist = function playlist(list) {
+    if (list) {
+      plist = list.slice();
+    }
+
+    return plist.slice();
+  };
+
+  playlist.currentItem = function item(index) {
+    if (typeof index === 'number' && index >= 0 && index < plist.length) {
+      currentIndex = index;
+      playItem(player, autoadvanceTimeout, plist[currentIndex]);
+      return currentIndex;
+    }
+
+    return currentIndex;
+  };
+
+  playlist.next = function next() {
+    var prevIndex = currentIndex;
+    currentIndex = Math.min(currentIndex + 1, plist.length - 1);
```

Comment:
Simply for readability, might be worth setting `plist.length - 1` to a variable like `lastIndex`. A comment would work too, but at a glance this looks more complex than it needs to.